testcontainers-dapr/src/main/java/io/dapr/testcontainers/DaprContainer.java

```diff
@@ -0,0 +1,447 @@
+/*
+ * Copyright 2024 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.testcontainers;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.images.builder.Transferable;
+import org.testcontainers.utility.DockerImageName;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.nodes.Tag;
+import org.yaml.snakeyaml.representer.Representer;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class DaprContainer extends GenericContainer<DaprContainer> {
+
+  public enum DaprLogLevel {
+    ERROR,
+    WARN,
+    INFO,
+    DEBUG
+  }
+
+  public static class Subscription {
+    String name;
+    String pubsubName;
+    String topic;
+    String route;
+
+    /**
+     * Creates a new subscription.
+     * @param name Subscription name.
+     * @param pubsubName PubSub name.
+     * @param topic Topic name.
+     * @param route Route.
+     */
+    public Subscription(String name, String pubsubName, String topic, String route) {
+      this.name = name;
+      this.pubsubName = pubsubName;
+      this.topic = topic;
+      this.route = route;
+    }
+  }
+
+  public static class MetadataEntry {
+    String name;
+    Object value;
+
+    public MetadataEntry(String name, Object value) {
+      this.name = name;
+      this.value = value;
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public void setName(String name) {
+      this.name = name;
+    }
+
+    public Object getValue() {
+      return value;
+    }
+
+    public void setValue(String value) {
+      this.value = value;
+    }
+  }
+
+  /**
+   * Represents a Dapr component.
+   */
+  public static class Component {
+    String name;
+
+    String type;
+
+    String version;
+
+    List<MetadataEntry> metadata;
+
+    /**
+     * Creates a new component.
+     * @param name Component name.
+     * @param type Component type.
+     * @param version Component version.
+     * @param metadata Metadata.
+     */
+    public Component(String name, String type, String version, Map<String, Object> metadata) {
```

Comment:
Please, move the domains objects to their own files. It pollutes this file too much IMO.

Reply:
done